### ISSUE TITLE
Update never completes (ever)

### ISSUE THREAD

**Issue by Stono** [terraform-providers/terraform-provider-google#1041]
Hey,
Take this example of me making a single change to my cluster:

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ module.kubernetes.google_container_cluster.cluster
      min_master_version: "1.8.6-gke.0" => "1.8.7-gke.0"


Plan: 0 to add, 1 to change, 0 to destroy.

```

You'll see from the debug logs that terraform never actually stops (this is only a couple of minutes, but i've left it for 20+):
```
2018/02/02 20:14:46 [DEBUG] apply: google_container_cluster.cluster: executing Apply
2018-02-02T20:14:46.673Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:46 [DEBUG] Locking "google-container-cluster/at-delivery-platform-dev/europe-west1-c/delivery-platform"
2018-02-02T20:14:46.673Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:46 [DEBUG] Locked "google-container-cluster/at-delivery-platform-dev/europe-west1-c/delivery-platform"
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:46 [DEBUG] Google API Request Details:
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: ---[ REQUEST ]---------------------------------------
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: PUT /v1/projects/at-delivery-platform-dev/zones/europe-west1-c/clusters/delivery-platform?alt=json HTTP/1.1
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Host: container.googleapis.com
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: User-Agent: google-api-go-client/0.5 (linux amd64) Terraform/0.11.3-dev
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Content-Length: 542
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Content-Type: application/json
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Accept-Encoding: gzip
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:46.681Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: -----------------------------------------------------
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:47 [DEBUG] Google API Response Details:
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: ---[ RESPONSE ]--------------------------------------
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: HTTP/2.0 200 OK
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Alt-Svc: hq=":443"; ma=2592000; quic=51303431; quic=51303339; quic=51303338; quic=51303337; quic=51303335,quic=":443"; ma=2592000; v="41,39,38,37,35"
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Cache-Control: private
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Content-Type: application/json; charset=UTF-8
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Date: Fri, 02 Feb 2018 20:14:40 GMT
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Server: ESF
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Vary: Origin
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Vary: X-Origin
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Vary: Referer
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: X-Content-Type-Options: nosniff
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: X-Frame-Options: SAMEORIGIN
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: X-Xss-Protection: 1; mode=block
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: {
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "name": "operation-1517602480628-53585271",
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "zone": "europe-west1-c",
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "operationType": "UPDATE_CLUSTER",
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "status": "RUNNING",
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "selfLink": "https://container.googleapis.com/v1/projects/149341795397/zones/europe-west1-c/operations/operation-1517602480628-53585271",
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "targetLink": "https://container.googleapis.com/v1/projects/149341795397/zones/europe-west1-c/clusters/delivery-platform",
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "startTime": "2018-02-02T20:14:40.628100336Z"
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: }
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: -----------------------------------------------------
2018-02-02T20:14:47.081Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:47 [DEBUG] Waiting for state to become: [DONE]
2018-02-02T20:14:47.082Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:47 [DEBUG] Google API Request Details:
2018-02-02T20:14:47.083Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: ---[ REQUEST ]---------------------------------------
2018-02-02T20:14:47.083Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: GET /v1/projects/at-delivery-platform-dev/zones/europe-west1-c/operations/operation-1517602480628-53585271?alt=json HTTP/1.1
2018-02-02T20:14:47.083Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Host: container.googleapis.com
2018-02-02T20:14:47.083Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: User-Agent: google-api-go-client/0.5 (linux amd64) Terraform/0.11.3-dev
2018-02-02T20:14:47.083Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Accept-Encoding: gzip
2018-02-02T20:14:47.083Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:47.083Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:47.083Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: -----------------------------------------------------
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:47 [DEBUG] Google API Response Details:
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: ---[ RESPONSE ]--------------------------------------
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: HTTP/2.0 200 OK
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Alt-Svc: hq=":443"; ma=2592000; quic=51303431; quic=51303339; quic=51303338; quic=51303337; quic=51303335,quic=":443"; ma=2592000; v="41,39,38,37,35"
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Cache-Control: private
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Content-Type: application/json; charset=UTF-8
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Date: Fri, 02 Feb 2018 20:14:40 GMT
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Server: ESF
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Vary: Origin
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Vary: X-Origin
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Vary: Referer
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: X-Content-Type-Options: nosniff
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: X-Frame-Options: SAMEORIGIN
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: X-Xss-Protection: 1; mode=block
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: {
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "name": "operation-1517602480628-53585271",
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "zone": "europe-west1-c",
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "operationType": "UPDATE_CLUSTER",
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "status": "RUNNING",
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "selfLink": "https://container.googleapis.com/v1/projects/149341795397/zones/europe-west1-c/operations/operation-1517602480628-53585271",
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "targetLink": "https://container.googleapis.com/v1/projects/149341795397/zones/europe-west1-c/clusters/delivery-platform",
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "startTime": "2018-02-02T20:14:40.628100336Z"
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: }
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: -----------------------------------------------------
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:47 [DEBUG] Progress of operation "operation-1517602480628-53585271": "RUNNING"
2018-02-02T20:14:47.141Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:47 [TRACE] Waiting 2s before next try
2018-02-02T20:14:49.145Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:49 [DEBUG] Google API Request Details:
2018-02-02T20:14:49.146Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: ---[ REQUEST ]---------------------------------------
2018-02-02T20:14:49.146Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: GET /v1/projects/at-delivery-platform-dev/zones/europe-west1-c/operations/operation-1517602480628-53585271?alt=json HTTP/1.1
2018-02-02T20:14:49.146Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Host: container.googleapis.com
2018-02-02T20:14:49.146Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: User-Agent: google-api-go-client/0.5 (linux amd64) Terraform/0.11.3-dev
2018-02-02T20:14:49.146Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Accept-Encoding: gzip
2018-02-02T20:14:49.146Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:49.146Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:49.146Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: -----------------------------------------------------
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:49 [DEBUG] Google API Response Details:
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: ---[ RESPONSE ]--------------------------------------
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: HTTP/2.0 200 OK
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Alt-Svc: hq=":443"; ma=2592000; quic=51303431; quic=51303339; quic=51303338; quic=51303337; quic=51303335,quic=":443"; ma=2592000; v="41,39,38,37,35"
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Cache-Control: private
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Content-Type: application/json; charset=UTF-8
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Date: Fri, 02 Feb 2018 20:14:42 GMT
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Server: ESF
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Vary: Origin
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Vary: X-Origin
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: Vary: Referer
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: X-Content-Type-Options: nosniff
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: X-Frame-Options: SAMEORIGIN
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: X-Xss-Protection: 1; mode=block
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: {
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "name": "operation-1517602480628-53585271",
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "zone": "europe-west1-c",
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "operationType": "UPDATE_CLUSTER",
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "status": "DONE",
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "selfLink": "https://container.googleapis.com/v1/projects/149341795397/zones/europe-west1-c/operations/operation-1517602480628-53585271",
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "targetLink": "https://container.googleapis.com/v1/projects/149341795397/zones/europe-west1-c/clusters/delivery-platform",
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "startTime": "2018-02-02T20:14:40.628100336Z",
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:   "endTime": "2018-02-02T20:14:41.11346657Z"
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: }
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4:
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: -----------------------------------------------------
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:49 [DEBUG] Progress of operation "operation-1517602480628-53585271": "DONE"
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:49 [INFO] GKE cluster delivery-platform master authorized networks config has been updated
2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 2018-02-02T20:14:49.226Z [DEBUG] plugin.terraform-provider-google_v1.5.0_x4: 2018/02/02 20:14:49 [DEBUG] Locking "google-container-cluster/at-delivery-platform-dev/europe-west1-c/delivery-platform"
module.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 10s elapsed)
module.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 20s elapsed)
^[[Amodule.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 30s elapsed)
module.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 40s elapsed)
module.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 50s elapsed)
module.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 1m0s elapsed)
module.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 1m10s elapsed)
module.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 1m20s elapsed)
module.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 1m30s elapsed)
module.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 1m40s elapsed)
^[[A^[[Amodule.kubernetes.google_container_cluster.cluster: Still modifying... (ID: delivery-platform, 1m50s elapsed)
```

Interestingly it says DONE, but nothing has change (my master is still on the old version).
I also find it very strange that in the debug logs we have `GKE cluster delivery-platform master authorized networks config has been updated`.  Definitely didn't ask for that?!

**Comment by Stono** [comment]
The master authorised networks thing could be linked to https://github.com/terraform-providers/terraform-provider-google/issues/1029

**Comment by emilymye** [comment]
@Stono would you mind adding the config you're using to this issue and to #1029? It would very helpful for debugging purposes. I think I know what's happening here but the config would be very helpful for your other issue.

**Comment by Stono** [comment]
@emilymye sure:

main.tf
```
❯ cat terraform/modules/container-cluster/main.tf
# create a regional subnet for this cluster
resource "google_compute_subnetwork" "vpc_subnet" {
  name          = "${var.prefix}"
  network       = "${var.network}"
  project       = "${var.network_project}"
  region        = "${var.region}"
  ip_cidr_range = "${var.node_cidr_range}"
  private_ip_google_access = true
  secondary_ip_range = {
    range_name = "${var.prefix}-pods"
    ip_cidr_range = "${var.pod_cidr_range}"
  }
  secondary_ip_range = {
    range_name = "${var.prefix}-services"
    ip_cidr_range = "${var.service_cidr_range}"
  }
}

resource "null_resource" "k8-provision" {
  triggers {
    custom_pool_id = "${google_container_cluster.cluster.id}"
  }
  depends_on = ["google_container_cluster.cluster"]

  provisioner "local-exec" {
    command = <<EOT
      ${path.module}/configure_k8s.sh \
      ${google_container_cluster.cluster.name} \
      ${var.project} \
      ${var.primary_zone}
    EOT
    when = "create"
  }

  # This attempts to gracefully stop things running on the cluster
  provisioner "local-exec" {
    command = <<EOT
      ${path.module}/destroy_k8s.sh \
      ${google_container_cluster.cluster.name} \
      ${var.project} ${var.primary_zone}
    EOT
    on_failure = "continue"
    when = "destroy"
  }
}

resource "null_resource" "install-helm" {
  triggers {
    custom_pool_id = "${google_container_cluster.cluster.id}"
    helm_version = "2.7.2"
  }
  depends_on = ["google_container_cluster.cluster"]

  provisioner "local-exec" {
    command = <<EOT
      ${path.module}/configure_helm.sh \
        ${google_container_cluster.cluster.name} \
        ${var.project} \
        ${var.primary_zone} \
        ${null_resource.install-helm.triggers.helm_version}
    EOT
    when = "create"
  }
}

resource "google_container_node_pool" "custom-pool" {
  name       = "custom-pool"
  node_count = "${var.machine["min_nodes"]}"
  cluster    = "${google_container_cluster.cluster.name}"
  zone       = "${var.primary_zone}"

  lifecycle {
    ignore_changes = ["node_count"]
  }

  node_config {
    machine_type = "${var.machine["type"]}"
    disk_size_gb = "${var.machine["disk_size"]}"
    oauth_scopes = "${var.gke_node_scopes}"
    preemptible  = "${var.machine["preemptible"]}"

    labels {
      purpose = "default"
    }
  }

  autoscaling {
    min_node_count = "${var.machine["min_nodes"]}"
    max_node_count = "${var.machine["max_nodes"]}"
  }

  provisioner "local-exec" {
    command = <<EOT
      gcloud --quiet container node-pools delete default-pool \
        --zone ${google_container_cluster.cluster.zone} \
        --cluster ${google_container_cluster.cluster.name} \
        --project ${var.project}
    EOT
    when = "create"
  }
}

resource "google_container_cluster" "cluster" {
  name = "${var.prefix}"

  depends_on = ["google_compute_subnetwork.vpc_subnet"]

  # What zones should we run in
  zone = "${var.primary_zone}"
  additional_zones = "${var.additional_zones}"
  enable_legacy_abac = false

  lifecycle {
    ignore_changes = ["initial_node_count", "network_policy", "node_config", "addons_config"]
  }

  # This bit largely doesn't matter as we delete the default node pool anyway
  initial_node_count =  1
  node_config {
    machine_type = "${var.machine["type"]}"
    disk_size_gb = "100"
    oauth_scopes = "${var.gke_node_scopes}"
    preemptible  = false
  }

  # The whitelist ranges able to use kubectl
	# If you're editing this, please read:
	# https://github.com/terraform-providers/terraform-provider-google/issues/1029
  master_authorized_networks_config {
    cidr_blocks = [
      {
        cidr_block = "193.243.131.243/32",
        display_name = "Manchester (DC2)"
      },
      {
        cidr_block = "193.243.130.243/32",
        display_name = "Manchester (DC1)"
      },
      {
        cidr_block = "193.243.130.172/32",
        display_name = "Cloudstack (TCY)"
      },
      {
        cidr_block = "193.243.131.172/32",
        display_name = "Cloudstack (GTC)"
      },
      {
        cidr_block = "109.159.110.43/32",
        display_name = "Karl (Home)"
      },
      {
        cidr_block = "94.195.34.243/32",
        display_name = "Meves (Home)"
      },
      {
        cidr_block = "80.3.154.28/32",
        display_name = "Darren (Home)"
      }
    ]
  }

  # Enable network policy for custom rules
  network_policy {
    enabled = false
    provider = "CALICO"
  }

  # Network configuration
  network = "projects/${var.network_project}/global/networks/${var.network}"
  subnetwork = "${google_compute_subnetwork.vpc_subnet.name}"

  # Logging and monitoring configuration
  monitoring_service = "none"
  logging_service = "none"

  # To check available versions
  # Go to: gcloud container get-server-config --zone=europe-west2-a
  min_master_version = "${var.version}"
  node_version = "${var.version}"

  # Add on configuration
  addons_config {
    http_load_balancing {
      # We don't need this as we're using istio and ingress loadbalancer
      disabled = true
    }
    horizontal_pod_autoscaling {
      # Enable the horizontal pod autoscaler addon so we can do autoscaling
      disabled = true
    }
    kubernetes_dashboard {
      # Should we run a k8 dashboard
      disabled = true
    }
  }

  ip_allocation_policy {
    cluster_secondary_range_name = "${var.prefix}-pods"
    services_secondary_range_name = "${var.prefix}-services"
  }

  timeouts {
    create = "10m"
    delete = "30m"
    update = "2h"
  }
}
```

variables:
```
❯ cat terraform/modules/container-cluster/variables.tf
variable "node_cidr_range" {
  description = "The ip range for the subnet which kubernetes machines will be created in"
}

variable "pod_cidr_range" {
  description = "The range that the containers should be allocated from"
}

variable "service_cidr_range" {
  description = "The range to allocate cluster services from"
}

variable "version" {
  description = "The kubernetes version to use"
  default = "1.8.7-gke.0"
}

variable "project" {}
variable "region" {}
variable "primary_zone" {}

variable "machine" {
  type = "map"
  description = "What is the machine specification"

  default = {
    type = "n1-standard-4"
    disk_size = 100
    min_nodes = 1
    max_nodes = 2
    preemptible = false
  }
}

variable "additional_zones" {
  type = "list"
  description = "Do we want this to be a multi zone cluster?"
  default = []
}

# These are the scopes that the node pools run under
variable "gke_node_scopes" {
  type = "list"
  description = "The GKE node scopes"
}

variable "prefix" {
  description = "The prefix to use when building resources"
}

variable "network" {
  description = "The network to deploy the subnet into"
}

variable "network_project" {
  description = "The project that the main network resides in, and where to create the subnet"
}
```

And then the code which calls the above module:
```
locals {
  # See: https://cloud.google.com/kubernetes-engine/docs/how-to/ip-aliases
  cluster_service_cidr_range = "10.192.0.0/20"    # 10.192.0.0 - 10.192.15.255
  cluster_node_cidr_range = "10.193.0.0/23"       # 10.193.0.0 - 10.193.1.255
  cluster_pod_cidr_range = "10.194.0.0/15"        # 10.194.0.0 - 10.195.255.255
  region = "europe-west1"
  primary_zone = "europe-west1-c"
  additional_zones = ["europe-west1-d"]
  prefix = "delivery-platform"
}

# These are the scopes that the node pools run under
variable "gke_node_scopes" {
  type = "list"
  default = [
    "https://www.googleapis.com/auth/compute",
    "https://www.googleapis.com/auth/devstorage.read_write",
    "https://www.googleapis.com/auth/logging.write",
    "https://www.googleapis.com/auth/monitoring"
  ]
}

module "kubernetes" {
  source = "./modules/container-cluster"
  prefix = "${local.prefix}"
  project = "${var.project}"
  region = "${local.region}"
  primary_zone = "${local.primary_zone}"
  gke_node_scopes = "${var.gke_node_scopes}"
  additional_zones = "${local.additional_zones}"

  # The module will create a subnet in the given region for you
  # and create named networks for the passed CIDR ranges
  network = "${google_compute_network.vpc.name}"
  network_project = "${var.project}"

  node_cidr_range = "${local.cluster_node_cidr_range}"
  pod_cidr_range = "${local.cluster_pod_cidr_range}"
  service_cidr_range = "${local.cluster_service_cidr_range}"

  machine = {
    type = "custom-6-12288"
    disk_size = 200
    min_nodes = 1
    max_nodes = 5
    preemptible = "${var.preemptible}"
  }
}
```

**Comment by emilymye** [comment]
Hi @Stono, I'm merging in a fix for this issue. I'm going to cross-post your config in #1029 but can you also verify that #1029 is still happening for you after the fix (#1042)?
